pkg/api-manager/istio/istio.go

```diff
@@ -0,0 +1,91 @@
+package istio
+
+import (
+	"context"
+	"strings"
+
+	"istio.io/istio/pilot/pkg/config/kube/crd"
+	"istio.io/istio/pilot/pkg/model"
+
+	ewrapper "github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vmware/dispatch/pkg/trace"
+)
+
+const (
+	// InternalGateway represents the gateway that we should redirect to
+	// Should probably come from a config file
+	InternalGateway = "knative-ingressgateway.istio-system.svc.cluster.local"
+)
+
+func NewClient() (*Client, error) {
+	log.Infoln("Trying to create istio client")
+	crdClient, err := crd.NewClient("", "", []model.ProtoSchema{model.VirtualService}, "")
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		istioClient: crdClient,
+	}, nil
+}
+
+type Client struct {
```

Comment:
You don't really need another layer of istio client abstraction: there's already an API client that you're using here in the `istioClient` field.